### ISSUE TITLE
How can we assign variables with `if` statement?

### ISSUE THREAD

**Issue by ceremcem** [pyos/dg#39]
It's both useful to be able to assign variables with `if` statements: 

```
a = 4
x = if a > 5 => 2
    otherwise => 4 
```
and makes function returns more consistent: 

```
my_func = x ->
    if x > 5 =>
        None
    otherwise =>
        'hey'


print "expecting None: " <|  my_func 6
print "expecting 'hey': " <|  my_func 3
```

![image](https://user-images.githubusercontent.com/6639874/28493365-50711fc6-6f1e-11e7-9ede-d2c513032258.png)

How could it be possible to assign variables in this way?

**Comment by pyos** [comment]
the assignment works as expected (x becomes 4), so i'm not sure what the question is about. if you're typing this into the repl, note that it considers `x = if a > 5 => 2` to be a complete expression (with an implied `otherwise => None`); you have to explicitly force it to wait for the second line by wrapping the whole construct into parentheses

```
>>> x = if a > 5 => 2
>>>     otherwise => 4
4
>>> print x
None
>>> (
... x = if a > 5 => 2
...     otherwise => 4
... )
>>> print x
4
```

as for `my_func`, the indentation doesn't work that way. `if` is a built-in "function" that expects `condition => action` expressions as arguments, with a special hack in the parser that overrides operator priorities in order to make `if x => y` on a single line parse as `if (x => y)` rather than the normal behavior of `f x => y` == `(f x) => y`. and then there's a rule that if a function call is followed by an indented block, each statement in that block is a separate argument to the call. thus valid uses of `if` must look like this:

```
if c1 => a1
   c2 => a2 # note that the whole argument, including the condition, is indented relative to `if`
   c3 => a3
   ...
```

where `c1 => a1` is optional, but if present, `a1` must fit onto a single line or be parenthesized - only `a2` and so on may be indented blocks without enclosing parentheses, else there would be an ambiguity because

```
if c1 =>
  a1
```

may be parsed as either `if (c1 =>) (a1)` or `if (c1 => (a1))` (it happens to be the latter due to what is probably a bug in the parser).

tl;dr:

```
my_func = x ->
  if x > 5 => None
     otherwise => 'hey'
```

or

```
my_func = x -> if
  x > 5 =>
    None
  otherwise =>
    'hey'
```

**Comment by ceremcem** [comment]
Hmm, it makes sense now. 

Yes, I was trying to test the first code in REPL, that's why it didn't work (because of not using the parenthesis)

Following code works as expected: 

```dg

my_func = x ->
    if
        x > 5 =>
            None
        otherwise =>
            'hey'


print "expecting None: " <|  my_func 6
print "expecting 'hey': " <|  my_func 3

x = 6

a = if
    x > 5 =>
        'hello'
    otherwise =>
        'there'


print "expecting 'hello': " <|  a

a = 4
x = if a > 5 => 2
    otherwise => 4

print "expecting 4: " <| x
```

![image](https://user-images.githubusercontent.com/6639874/28495622-910fc940-6f5c-11e7-8fd0-4afccebe2254.png)


As far as I could understand, `if` works much like `switch` statement, so it requires that one level more indentation for above usage. Okay, it doesn't hurt.
